packages/kbn-generate-csv/src/generate_csv.ts

```diff
@@ -356,6 +356,9 @@ export class CsvGenerator {
         // set the latest pit, which could be different from the last request
         searchSource.setField('pit', { id: pitId, keep_alive: settings.scroll.duration });
 
+        // TODO: should we skip `_doc` (the tie breaker from Discover) from searchSource `sort` field when generating CSV?
+        // According to docs, the default `_shard_doc` is a better option when PIT is used.
+
         const results = await this.doSearch(searchSource, settings, searchAfter);
```

Comment:
@jughosta QQ: we don't store the `_doc` field in out searchSource, we just use `getSortForSearchSource` for getting the searchSource we send to Elasticsearch, right? So why do we need to remove it, or am I overlooking something and _doc is added the the persisted saved search? thx!

Reply:
@kertal I am applying some changes to skip it for other cases besides Discover page and embeddable.

Reply:
makes sense, thx!